src/fparser/two/tests/utils/test_walk.py

```diff
@@ -93,3 +95,26 @@ def test_walk_debug(capsys):
     stdout, _ = capsys.readouterr()
     assert stdout.startswith(8 * " " + "child type =")
     assert "Program" not in stdout
+
+def test_reproduce_issue367():
```

Comment:
Please could you rename this test to something like `test_walk_tuple` and add a brief doc string to the test.
This file also has a couple of Black formatting errors.
Once that's done we really should be ready to go :-)